### PR TITLE
[autoscaler] fix AWS example.yaml

### PR DESCRIPTION
**Pull request by pschafhalter**
## What do these changes do?

- introduce default setting for availability_zone configuration option
- set AWS example.yaml's availability_zone to default

## Related issue number

- #1407

### CODE REVIEW

File: python/ray/autoscaler/aws/example.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ idle_timeout_minutes: 5
 provider:
     type: aws
     region: us-west-2
-    # availability_zone: us-west-2a
+    availability_zone: default
```
Comment (robertnishihara) [inline python/ray/autoscaler/aws/example.yaml]: Will this work if `availability_zone` is not provided? I don't think it should be required.

If it is required, then you'll also need to change `development_example.yaml`.

Comment (yaroslavvb) [reply python/ray/autoscaler/aws/example.yaml]: PS: some availability zones may not contain any instances of desired type. Perhaps t2/c5 may be in all instances, but I've seen error this with other instance types

Comment (yaroslavvb) [reply python/ray/autoscaler/aws/example.yaml]: You get error like this `botocore.exceptions.ClientError: An error occurred (Unsupported) when calling the RunInstances operation: Your requested instance type (g3.4xlarge) is not supported in your requested Availability Zone (us-east-1f). Please retry your request by not specifying an Availability Zone or choosing us-east-1b, us-east-1a, us-east-1c, us-east-1e.`

**Review by ericl — APPROVED** [review]
Pushed some fixes. I will followup with another PR to create a default subnet if not found.

**Review by pcmoritz — CHANGES_REQUESTED** [review]
This doesn't seem to work for me, it needs to be availability_zone instead of availability-zone; maybe it depends on the version of pyyaml or something, but using an underscore should always be safe.

@ericl Can you change it?

File: python/ray/autoscaler/aws/development-example.yaml [hunk 2]
```diff
@@ -22,6 +22,7 @@ idle_timeout_minutes: 5
 provider:
     type: aws
     region: us-west-2
+    availability-zone: us-west-2a
```
Comment (robertnishihara) [inline python/ray/autoscaler/aws/development-example.yaml]: Can we get rid of this line? I'd prefer to not specify the availability zone in the development example (if only to verify that it works without specifying it).

Comment (pschafhalter) [reply python/ray/autoscaler/aws/development-example.yaml]: Getting rid of the line results in an error parsing the YAML. I can re-implement the "default" option, or I can modify the YAML parser to make "availability-zone" optional.

Comment (robertnishihara) [reply python/ray/autoscaler/aws/development-example.yaml]: Yes, this is the bug reported in #1407. The availability zone should not be required (and the user shouldn't have to say say `default` either).

**Comment by robertnishihara** [comment]
Closing for now since it has been a while. Please reopen if this is still in progress.

**Pull request closed without merging**